### PR TITLE
Move lmnet/executor/train.py to blueoil/cmd

### PR DESCRIPTION
**Pull request by tfujiwar**
## What this patch does to fix the issue.
- Remove blueoil/cmd/train.py (https://github.com/blue-oil/blueoil/commit/77d102a5f938b60ac8db1e688fb44d376839ba32)
- Move lmnet/executor/train.py to blueoil/cmd (https://github.com/blue-oil/blueoil/commit/d0dbd45c7af5f06e8ee08e49d98062a71546ff7e)
- Update blueoil/cmd/train.py (https://github.com/blue-oil/blueoil/commit/749daf900d52969ed1909a63f23905e37e9e7175)

This PR looks big but most of the changes are just copied. Maybe it's easier to check each commit one by one than checking the whole changes.

## Link to any relevant issues or pull requests.
#746

### CODE REVIEW

File: blueoil/cmd/main.py [hunk 1]
```diff
@@ -68,10 +68,27 @@ def init(output):
...
+@click.option(
+    '--recreate',
+    is_flag=True,
+    help='Delete and recreate experiment id dir',
+    default=False,
+)
+@click.option(
+    '-n',
+    '--network',
+    help='Network name which you want to use for this training. override config.DATASET_CLASS',
```
Comment (iizukak) [inline blueoil/cmd/main.py]: I don't know why we need this option.
Specify in config file is not enough?

File: blueoil/cmd/main.py [hunk 2]
```diff
@@ -68,10 +68,27 @@ def init(output):
...
+    default=None,
+)
+@click.option(
+    '-d',
+    '--dataset',
+    help='Dataset name which is the source of this training. override config.NETWORK_CLASS',
```
Comment (iizukak) [inline blueoil/cmd/main.py]: Same question.

File: blueoil/cmd/train.py [hunk 3]
```diff
@@ -13,39 +13,378 @@
...
+            DatasetClass = TFDSObjectDetection
+        else:
+            DatasetClass = TFDSClassification
+
+    dataset = DatasetClass(subset=subset, **dataset_kwargs, **tfds_kwargs)
+    enable_prefetch = dataset_kwargs.pop("enable_prefetch", False)
+    return DatasetIterator(dataset, seed=rank, enable_prefetch=enable_prefetch)
+
+
+def start_training(config):
```
Comment (iizukak) [inline blueoil/cmd/train.py]: This function is too long, But it's copied from original `train.py`.
We will refactor in other PR.

File: blueoil/cmd/main.py (hunk 1 above)
Comment (tfujiwar) [reply blueoil/cmd/main.py]: I also don't know why but I think we can remove this.

Comment (tfujiwar) [reply blueoil/cmd/main.py]: I've removed the options `network` and `dataset`.

**Pull request merged**